sync3/handler/connstate.go

```diff
@@ -434,6 +440,35 @@ func (s *ConnState) buildRooms(ctx context.Context, builtSubs []BuiltSubscriptio
 	return result
 }
 
+func (s *ConnState) lazyLoadTypingMembers(ctx context.Context, response *sync3.Response) {
+	for roomID, typingEvent := range response.Extensions.Typing.Rooms {
+		if !s.lazyCache.IsLazyLoading(roomID) {
+			continue
+		}
+		room, ok := response.Rooms[roomID]
+		if !ok {
+			room = sync3.Room{}
+		}
+		typingUsers := gjson.GetBytes(typingEvent, "content.user_ids")
+		for _, typingUserID := range typingUsers.Array() {
+			if s.lazyCache.IsSet(roomID, typingUserID.Str) {
+				// client should already know about this member
+				continue
+			}
+			// load the state event
+			memberEvent := s.globalCache.LoadStateEvent(ctx, roomID, s.loadPosition, "m.room.member", typingUserID.Str)
+			if memberEvent != nil {
+				room.RequiredState = append(room.RequiredState, memberEvent)
+				s.lazyCache.AddUser(roomID, typingUserID.Str)
+			}
+		}
+		// only add the room if we have membership events
+		if len(room.RequiredState) > 0 {
+			response.Rooms[roomID] = room
+		}
```

Comment:
Confused by the comment. It may be the case that

- RequiredState was nonempty before,
- we had no memberships events to add,
- but we still write the `room` back to the map (which will be a no-op).

The behaviour is fine, it's just the wording doesn't seem to match.

Reply:
Oh, this isn't new code, it's just moved from ConnStateLive.

Reply:
Yup.